docs/design-docs/01-kpt-live-stdin.md

```diff
@@ -0,0 +1,242 @@
+# Title
+
+* Author(s): Yuwen Ma <yuwenma@google.com>
+* Contributor(s):
+    - Sean Sullivan <seans@google.com>
+    - Sunil Arora <sunilarora@google.com>
+    - Mortent Torkildsen <mortent@google.com>
+    - Mengqi Yu <mengqiy@google.com>
+    - Natasha Sarkar <natashasarkar@google.com>
+
+* Approver: Sunil Arora <sunilarora@google.com>
+
+## Why
+
+This design is aimed at making `kpt live` be hydration-tool-neutral so that `kpt live`
+applicable use cases can be expanded to `kustomize build`, `helm templates` and raw
+kubernetes manifests.
+
+### Non-goal
+- This doc is not aimed at designing a unified hydration between kpt and kustomize
+- This doc is not aimed at changing the kpt in-place hydration models.
+
+## Design
+
+We propose to make `kpt live` support reading configurations from standard input,
+detach the Inventory from Kptfile and simplify the Inventory-ResourceGroup mechanism
+to use  ResourceGroup only.
+
+### Read standard input resources
+
+We will add a reserved keyword “-” as a special kpt directory. This "-" accepts resources
+from the standard input. For example,  `kustomize build | kpt live apply -`
+
+### Initialize a `ResourceGroup` object
+
+`kpt live init` will create a ResourceGroup CR in resourcegroup.yaml. Only three fields
+should be given to the CR.
+
+- metadata.name: A client-provided valid RFC 1123 DNS subdomain name. Default value has prefix “inventory-” with
```

Comment:
What is rationale for appending this 8 digit number ? (when one specifiy it from the commandline then the number will still be appended or not ?) Also, is this related to the `inventory-id` ?

Reply:
this is just random number, I guess it's to avoid name duplication. If user provides the "name", it will use what user specified. 
Not related to inventory-id.